docs/UserGuide.md

```diff
@@ -39,136 +39,281 @@ AddressBook Level 3 (AB3) is a **desktop app for managing contacts, optimized fo
+| id        | Integer number        | Must be an integer as IDs are unique identifiers for each customer.         |
+
+#### What to Expect
+- **If Successful:** 
+  - Message: "Here are all the customers that match your search: (List of customers)."
+- **If There is an Error:** 
+  - Missing field: "Please provide a name/job/id." (Specific to the flag used)
+  - Invalid or unsupported flag: "Please provide a valid flag."
+  - No matching customers: "No customers were found. Please check the name/job/id." (Specific to the flag)
+
+**Handling Duplicates:**  
+- For `name` and `job` fields: Multiple customers can have the same names or jobs. The command will list all matching entries.
+- For the `id` field: Since customer IDs are intended to be unique, finding multiple customers with the same ID will
+  trigger a warning: 
+  - "There seems to be multiple customers tagged to the same ID, use the delete command to remove all customers with the affected ID, and re-add customers accordingly."
+---
 
-### Exiting the program : `exit`
+### Feature 6: Save Remarks About Customers
+
+**Purpose:**  
+Allows users to save specific notes or remarks about a customer, which can be viewed later to recall notable details.
+
+**How to Use It:**  
+- **Command Format:** 
+```
+note <customer id> <note here>
+```
+- **Example:** 
+```
+note 55 He is a problematic customer.
+```
+
+#### Parameters
+| Parameter | Expected Format       | Explanation                                                              |
+|-----------|-----------------------|--------------------------------------------------------------------------|
+| ID        | Integer (0 to the last ID) | Ensures the note is attached to a valid customer ID.                     |
+| Note      | Any string            | Notes are case-insensitive and can include any textual information.      |
+
+#### What to Expect
+- **If Successful:** 
+  - Message: "Note has been added to Customer `<Customer ID>`."
+- **If There is an Error:** 
+  - Invalid ID: "No customer with `<Customer ID>` exists. Please input a valid ID."
+
+**Handling Duplicates:**  
+- Although customer IDs should be unique, in the rare case where duplicates are detected, the following error message will be shown:
+  - "Sorry, it appears that multiple customers with id: `<Customer ID>` exist. Please use the delete command to remove the duplicated customer ID."
 
-Exits the program.
+---
 
-Format: `exit`
+### Feature 7: Add/Replace Credit Card Tier
+
+**Purpose:**  
+Allows users to assign or update the credit card tier for a customer. This is particularly useful for managing new, existing, or returning customers who may not have been assigned a credit card tier initially or who need their current tier updated.
+
+**How to Use It:**  
+- **Command Format:** 
+```
+tag <customer id> <tier>
+```
+- **Example:** 
+```
+tag 69 reject
```

Comment:
I like this idea, but we will have to refactor the tags now to mean credit card tiers.